common/lib/dependabot/clients/bitbucket.rb

```diff
@@ -10,12 +10,27 @@ class NotFound < StandardError; end
       class Unauthorized < StandardError; end
       class Forbidden < StandardError; end
 
+      #######################
+      # Constructor methods #
+      #######################
+
+      def self.for_source(source:, credentials:)
+        credential = credentials.
+                     find do |cred|
+          cred["type"] == "git_source" &&
+            cred["host"] == source.hostname
+        end
+
+        new(source, credential)
+      end
+
       ##########
       # Client #
       ##########
 
-      def initialize(credentials:)
-        @credentials = credentials
+      def initialize(source, credential)
```

Comment:
This seems to break calls for existing methods (for instance the file fetcher in the generic script).
I think to keep compatibility, it should be something more like
```
def initialize(credentials:, source: nil)
```